test/api/search_test.cpp

```diff
@@ -8,20 +8,22 @@
 TEST(get_overlap_records, simulated_chr1_small_golden)
 {
     // Construct Tree
-    std::vector<std::vector<bamit::Record>> records{};
     std::filesystem::path input{DATADIR"simulated_mult_chr_small_golden.bam"};
-    bamit::parse_file(input, records);
-
-    std::unique_ptr<bamit::IntervalNode> root(nullptr);
-    bamit::construct_tree(root, records);
+    bamit::sam_file_input_type input_file{input};
+    std::vector<std::unique_ptr<bamit::IntervalNode>> node_list = bamit::construct_tree(input_file);
 
     // Search Tree
     std::filesystem::path tmp_dir = std::filesystem::temp_directory_path();     // get the temp directory
     std::filesystem::path result_sam_path{tmp_dir/"result.sam"};
-    bamit::sam_file_input_type input_bam{input};
     bamit::Position start{1, 100};
     bamit::Position end{1, 110};
-    get_overlap_records(input_bam, root, start, end, result_sam_path);
+    // Potentially undefined behavior in our seqan branch: input_file is parsed in the tree construction,
+    // so using the same input_file variable here leads to incorrect results (probably because the iterator no longer
+    // points to the beginning of the file).
+    // Temporary fix: use new object.
+    // Permanent fix: seqan3 is changing how you get/use the file position.
```

Comment:
Is the test file the best place to mention this problem? Should be in the include file and maybe add a #TODO or something?